nx/lib/nx/binary_backend/matrix.ex

```diff
@@ -546,11 +546,7 @@ defmodule Nx.BinaryBackend.Matrix do
       |> Enum.map(fn {row, idx} -> Enum.at(row, idx) end)
       |> Enum.reject(&is_nil/1)
 
-    {s, [vt_row | _] = vt} = apply_singular_value_corrections(s, vt)
-
-    if length(vt) != vt_rows or length(vt_row) != vt_cols do
```

Comment:
Seems like we lost this check during the refactor

Reply:
Whoops 🙈